### DISCUSSION TITLE
protected file read and write issues

### DISCUSSION THREAD

**Discussion by reclock** [gramineproject/gramine#1693] [Q&A] [answered]
There are currently some questions about the type of protected file , and my configuration is as follows:

**{path="/res_enc", uri="file:/home/res_enc/", type="encrypted", keyname="_sgx_mrsigner"},**

When I run the enclave application, the result file **res1. bin** will be generated in the **/home/res_enc/data1/** directory. If I manually move **res1. bin** to the **/home/res_enc/data2** path and fail to open the file, "**error: Permission denied**", is there a relationship between opening the ciphertext file and its path? Can you solve it by modifying something?

**Comment by dimakuv** [comment, marked as answer]
> is there a relationship between opening the ciphertext file and its path

Yes, there is a relationship between the encrypted file and its path. See https://www.tatetian.io/2017/01/15/understanding-sgx-protected-file-system/. This is done for security reasons.

> Can you solve it by modifying something?

Why do you need to move the file exactly?

What you can do is to emulate file move via creating a new file (under `/home/res_enc/data2/`) and copying the contents of your original file (under `/home/res_enc/data1/`) into this new file. Then Gramine does the correct re-encryption.

**Reply by reclock** [reply]
Because there is currently a scenario where training, deployment, and prediction are carried out separately. The deployment phase will transfer the trained model files to the predicted route. Based on your answer, we are ready to use Gramine to copy files without using the system's mv

Thanks
